heroic-core/src/main/java/com/spotify/heroic/http/query/QueryResource.java

```diff
@@ -143,6 +177,28 @@ private void bindMetricsResponse(
                 r.getLimits()));
     }
 
+    private QueryMetrics parseQueryMetricsFrom(String queryString) {
+        QueryMetrics query = null;
+        try {
+            query = objectMapper.readValue(queryString, QueryMetrics.class);
+        } catch (IOException e) {
+            log.info("Failed to deserialize JSON to QueryMetrics object! \"" + queryString + "\"");
+            log.info("Error: " + e.toString());
```

Comment:
Logging is the wrong choice, just throw the exception raised here and the user should be notified.

I'm a bit suspicious of this approach that it would maintain all user interaction (specifically supply the user with correctly formatted `4xx` responses). You are probably better of serializing the `QueryMetrics` instead.